packages/lib-auth/src/client.js

```diff
@@ -0,0 +1,320 @@
+        'project_id': user.projectId
+      }
+    }
+    const registerResponse = await this._httpClient.post('/users', data)
+
+    // After the user is created, we need to sign in to actually get an access
+    // token. `panoptes-javascript-client` manages to do it without this step;
+    // there's no `set-cookie` header on the response that I can see, so I'm not
+    // actually sure how.
+    const credentials = {
+      login: user.login,
+      password: user.password,
+    }
+    return this.signIn(credentials)
+  }
+
+  /**
+   * Resets the client state to its initial state
+   *
+   * @private
+   * @returns {void}
+   */
+  _resetState () {
+    this._state = {
+      accessToken: '',
+      accessTokenCreatedAt: NaN,
+      accessTokenExpiresAt: NaN,
+      accessTokenExpiresIn: NaN,
+      refreshToken: '',
+      scope: ''
+    }
+  }
+
+  /**
+   * Resumes a session for a logged-in user.
+   *
+   * @async
+   * @param {string} [jwt] - A JWT to exchange for an access token. Allows the client to be used on the server side by extracting the JWT included in the cookies in the `req` object and passing it in here. If run on the browser without the JWT argument, it will try to retrieve it from document object.
+   * @returns {Promise} Resolves to the new access token if logged in, or `null` if logged out.
+   */
+  async resumeSession (jwt = this._getJWTFromBrowser()) {
+    if (jwt) {
+      const newAccessToken = await this._exchangeJWTForToken(jwt)
+      return newAccessToken
+    } else {
+      return null
+    }
+  }
+
+  /**
+   * Signs a user in
+   *
+   * @param {Object} credentials - The credentials for the user logging in.
+   * @param {string} credentials.login - The user's username or email address.
+   * @param {string} credentials.password - The user's password.
+   * @returns {Promise} Resolves to the new access token.
+   */
+  async signIn (credentials) {
+    if (this.isSignedIn()) {
+      await this.signOut()
+    }
+
+    const csrfToken = await this._getCSRFToken()
+    const data = {
+      'authenticity_token': csrfToken,
+      user: {
+        login: credentials.login,
+        password: credentials.password,
+        remember_me: true
+      }
+    }
+    const response = await this._httpClient.post('/users/sign_in', data)
+    const jwt = this._getJWTFromResponse(response)
+    return this.resumeSession(jwt)
+  }
+
+  /**
+   * Signs a user out.
+   *
+   * @async
+   * @returns {Promise} Resolves to `null`.
+   */
+  async signOut () {
+    const currentSession = await this.getAccessToken()
+
+    // If there's no session, we can't log out.
+    if (!currentSession) {
+      return null
+    }
+
+    const csrfToken = await this._getCSRFToken()
+    const config = {
+      headers: {
+        'x-csrf-token': csrfToken
+      }
+    }
+    await this._httpClient.delete('/users/sign_out', config)
+    this._resetState()
+    return null
+  }
```

Comment:
Does `signOut` need to return anything. In the specs, nothing is done with the return value of `signOut`. If there's a case where something should be done with it, then there should be a spec for that.

Reply:
Probably not, actually